autolamella/milling.py

```diff
@@ -1,127 +1,76 @@
+def milling(
         microscope,
         settings,
+        stage_settings,
         my_lamella,
-        prefix="IB_" + filename_prefix,
-        suffix="_0b-unaligned",
-    )
-    realign(microscope, image_unaligned, my_lamella.fiducial_image)
-    # Save the newly aligned image for the next alignment stage
-    my_lamella.fiducial_image = grab_images(
-        microscope,
-        settings,
-        my_lamella,  # can remove
-        prefix="IB_" + filename_prefix,
-        suffix="_1-aligned",
-    )
-    # Create and mill patterns
-    _upper_milling_coords(microscope, stage_settings, my_lamella)
-    if not demo_mode:
-        print("Milling pattern...")
-        microscope.imaging.set_active_view(2)  # the ion beam view
-        try:
-            microscope.patterning.run()
-        except ApplicationServerException:
-            logging.error("ApplicationServerException: could not mill!")
-    microscope.patterning.clear_patterns()
-    grab_images(
-        microscope,
-        settings,
-        my_lamella,  # can remove
-        prefix="IB_" + filename_prefix,
-        suffix="_2-after-upper-milling",
-    )
-    return microscope
-
-
-def lower_milling(
-    microscope,
-    settings,
-    stage_settings,
-    my_lamella,
-    filename_prefix="",
-    demo_mode=False,
+        pattern,  # "upper", "lower", "both"
+        filename_prefix="",
+        demo_mode=False,
 ):
     from autoscript_core.common import ApplicationServerException
     from autoscript_sdb_microscope_client.structures import StagePosition
 
+    # Sanity-check pattern parameter
+    if pattern not in ("upper", "lower", "both"):
+        raise ValueError(f"Invalid pattern type:\n"
+                         f"Should be \"upper\", \"lower\" or \"both\", not \"{pattern}\"")
+
     # Setup and realign to fiducial marker
     setup_milling(microscope, settings, stage_settings, my_lamella)
     tilt_in_radians = np.deg2rad(stage_settings["overtilt_degrees"])
-    microscope.specimen.stage.relative_move(StagePosition(t=+tilt_in_radians))
-    image_unaligned = grab_images(
-        microscope,
-        settings,
-        my_lamella,
-        prefix="IB_" + filename_prefix,
-        suffix="_3a-unaligned",
-    )
-    realign(microscope, image_unaligned, my_lamella.fiducial_image)
-    # Repeat realignment - in cases where the shift is large we find that
-    # the beam shift movement is less accuracte, so we make a second check
-    image_unaligned = grab_images(
-        microscope,
-        settings,
-        my_lamella,
-        prefix="IB_" + filename_prefix,
-        suffix="_3b-unaligned",
-    )
-    realign(microscope, image_unaligned, my_lamella.fiducial_image)
+    if pattern == "upper":
+        microscope.specimen.stage.relative_move(StagePosition(t=-tilt_in_radians))
+    elif pattern == "lower":
+        microscope.specimen.stage.relative_move(StagePosition(t=+tilt_in_radians))
+
+    # Realign three times
+    for abc in "abc":
```

Comment:
Hey @MoritzWM, we've noticed this section of the code repeatedly realigning to the fiducial marker - can you tell me why it's there? 
Donig this three times, *every* time it returns to this area seems like it'd end up with a high dose in this region, possibly degrading the cross shaped marker, potentially unnecessarily.

Reply:
Hey,
on our microscope, the first alignment was often a bit overshooting so I added this.
Maybe we could add an option like alignment_runs to the protocol stages? It could be beneficial to only realign once for the early steps and more often for the later ones, where precision becomes more important.
What do you think?

Reply:
I'll discuss with my colleagues, see if they've noticed anything similar